From 91f24b650eb4f69e0d5b98e85b0503f0ab9dbeb0 Mon Sep 17 00:00:00 2001
From: Mikhail Boldyrev <miboldyrev@gmail.com>
Date: Mon, 3 Feb 2020 13:38:42 +0300
Subject: [PATCH] fixed SOCI_CXX11 flag

---
 ports/soci/portfile.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ports/soci/portfile.cmake b/ports/soci/portfile.cmake
index 9ba44d8bf..e9caf45ac 100644
--- a/ports/soci/portfile.cmake
+++ b/ports/soci/portfile.cmake
@@ -31,7 +31,7 @@ vcpkg_configure_cmake(
     PREFER_NINJA
     OPTIONS
         -DSOCI_TESTS=OFF
-        -DSOCI_CXX_C11=ON
+        -DSOCI_CXX11=ON
         -DSOCI_LIBDIR:STRING=lib # This is to always have output in the lib folder and not lib64 for 64-bit builds
         -DLIBDIR:STRING=lib
         -DSOCI_STATIC=${SOCI_STATIC}
-- 
2.24.1

