diff --git a/scripts/toolchains/linux.cmake b/scripts/toolchains/linux.cmake
index 68d90470..7508cd46 100644
--- a/scripts/toolchains/linux.cmake
+++ b/scripts/toolchains/linux.cmake
@@ -17,8 +17,8 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
     string(APPEND CMAKE_C_FLAGS_RELEASE_INIT " ${VCPKG_C_FLAGS_RELEASE} ")
     string(APPEND CMAKE_CXX_FLAGS_RELEASE_INIT " ${VCPKG_CXX_FLAGS_RELEASE} ")
 
-    string(APPEND CMAKE_C_FLAGS_INIT " ${CMAKE_C_FLAGS} ")
-    string(APPEND CMAKE_CXX_FLAGS_INIT " ${CMAKE_CXX_FLAGS} ")
+    string(APPEND CMAKE_C_FLAGS_INIT " ${CMAKE_C_FLAGS} -fsanitize=$ENV{SANITIZER} ")
+    string(APPEND CMAKE_CXX_FLAGS_INIT " ${CMAKE_CXX_FLAGS} -fsanitize=$ENV{SANITIZER} ")
 
     string(APPEND CMAKE_SHARED_LINKER_FLAGS_INIT " ${VCPKG_LINKER_FLAGS} ")
     string(APPEND CMAKE_EXE_LINKER_FLAGS_INIT " ${VCPKG_LINKER_FLAGS} ")
